Content.Server/GameObjects/Components/Chemistry/MovespeedStatusComponent.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Shared.Movement.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.ViewVariables;
+using System;
+using System.Threading;
+
+namespace Content.Server.GameObjects.Components.Chemistry
+{
+    //TODO: refactor movement modifier component because this is a pretty poor solution
+    [RegisterComponent]
+    class MovespeedStatusComponent : Component, IMoveSpeedModifier
+    {
+        
+        public override string Name => "ChemicalMovementSpeedChangeStatus";
```

Comment:
This component's pretty dirty considering the name is not related to its class name at all. You should change the class name to reflect it's for metabolism on mobs.